frontend/public/components/secrets/create-secret/WebHookSecretForm.tsx

```diff
@@ -1,86 +1,48 @@
-        <label className="control-label co-required" htmlFor="webhook-secret-key">
-          {t('public~Webhook secret key')}
-        </label>
-        <div className="pf-v5-c-input-group">
-          <input
-            className="pf-v5-c-form-control"
-            id="webhook-secret-key"
-            data-test="secret-key"
-            type="text"
-            name="webhookSecretKey"
-            onChange={this.changeWebHookSecretkey}
-            value={this.state.stringData.WebHookSecretKey}
-            aria-describedby="webhook-secret-help"
-            required
-          />
-          <button
-            type="button"
-            onClick={this.generateWebHookSecret}
-            className="pf-v5-c-button pf-m-tertiary"
-            data-test="webhook-generate-button"
-          >
-            {t('public~Generate')}
-          </button>
-        </div>
-        <p className="help-block" id="webhook-secret-help">
-          {t('public~Value of the secret will be supplied when invoking the webhook.')}
-        </p>
-      </div>
-    );
-  }
-}
-
-export const WebHookSecretForm = withTranslation()(WebHookSecretFormWithTranslation);
-
-type WebHookSecretFormState = {
-  stringData: {
-    [key: string]: string;
+export const WebHookSecretForm: React.FC<SecretSubFormProps> = ({ onChange, stringData }) => {
+  const { t } = useTranslation();
+  const changeWebHookSecretkey = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newSecret = event.target.value;
+    onChange({ stringData: { ...stringData, WebHookSecretKey: newSecret }, base64StringData: {} });
   };
-};
 
-type WebHookSecretFormProps = {
-  onChange: Function;
-  onError: Function;
-  onFormDisable: Function;
-  secretType: string;
-  stringData: {
-    [key: string]: string;
+  const generateWebHookSecret = () => {
+    const newSecret = generateSecret();
+    onChange({ stringData: { ...stringData, WebHookSecretKey: newSecret }, base64StringData: {} });
   };
+
+  return (
+    <div className="form-group">
+      <label className="control-label co-required" htmlFor="webhook-secret-key">
+        {t('public~Webhook secret key')}
+      </label>
+      <div className="pf-v5-c-input-group">
+        <input
+          className="pf-v5-c-form-control"
+          id="webhook-secret-key"
+          data-test="secret-key"
+          type="text"
+          name="webhookSecretKey"
+          onChange={changeWebHookSecretkey}
+          value={stringData.WebHookSecretKey}
+          aria-describedby="webhook-secret-help"
+          required
+        />
+        <button
+          type="button"
+          onClick={generateWebHookSecret}
+          className="pf-v5-c-button pf-m-tertiary"
+          data-test="webhook-generate-button"
+        >
+          {t('public~Generate')}
+        </button>
+      </div>
+      <p className="help-block" id="webhook-secret-help">
+        {t('public~Value of the secret will be supplied when invoking the webhook.')}
+      </p>
+    </div>
```

Comment:
Not new, but we are using 'webhook secret key' terminology here, when in reality this form controls the *value* of the webhook secret. The key is a static string: "WebhookSecretKey". We should update the wording.